src/rx-query.ts

```diff
@@ -324,29 +323,24 @@ export class RxQueryBase<
             });
         }
 
-
         /**
          * run _ensureEqual() here,
          * this will make sure that errors in the query which throw inside of the RxStorage,
          * will be thrown at this execution context and not in the background.
          */
-        return _ensureEqual(this)
-            .then(() => firstValueFrom(this.$))
-            .then(result => {
-                if (!result && throwIfMissing) {
-                    throw newRxError('QU10', {
-                        collection: this.collection.name,
-                        query: this.mangoQuery,
-                        op: this.op
-                    });
-                } else {
-                    return result;
-                }
+        await _ensureEqual(this);
+        const result = await firstValueFrom(this.$);
+        if (!result && throwIfMissing) {
```

Comment:
lol nice, dunno why they chained promises in the original code

Reply:
i don't understand this change.. is it just rewriting with await instead of then?

Reply:
maybe there's a reason he didn't want to use an `async` function?

Reply:
My IDE "simplified" that for me.

Reply:
AFAIK there's no functional difference between chaining promises and async/await in JS. IMO using async/await reads and codes much nicer

Reply:
my bias is to not change stuff if there's no need. this is a fork we have to keep up to date, after all